CO2_Gadget_EINK.h

```diff
@@ -453,34 +469,45 @@ void showBatteryIcon(int32_t posX, int32_t posY, bool forceRedraw) {
         display.fillRect(posX, posY + 4, 2, 6, GxEPD_BLACK);
         display.drawRoundRect(posX + 2, posY, 27, 14, 3, GxEPD_BLACK);  // Battery outter rectangle
         // Serial.println("-->[EINK] Drawn round rect at " + String(posX + 2) + ", " + String(posY) + " with size 27x14");
+        newBatteryValue = 10;
         if (batteryLevel > 20) {
+            newBatteryValue = 20;
             display.fillRect(posX + 6, posY + 2, 4, 10, GxEPD_BLACK);
             // Serial.println("-->[EINK] Drawn rect at " + String(posX + 8) + ", " + String(posY + 2) + " with size 4x10");
         }
         if (batteryLevel > 40) {
+            newBatteryValue = 40;
             display.fillRect(posX + 11, posY + 2, 4, 10, GxEPD_BLACK);
             // Serial.println("-->[EINK] Drawn rect at " + String(posX + 14) + ", " + String(posY + 2) + " with size 4x10");
         }
         if (batteryLevel > 60) {
+            newBatteryValue = 60;
             display.fillRect(posX + 16, posY + 2, 4, 10, GxEPD_BLACK);
             // Serial.println("-->[EINK] Drawn rect at " + String(posX + 20) + ", " + String(posY + 2) + " with size 4x10");
         }
         if (batteryLevel > 80) {
+            newBatteryValue = 80;
             display.fillRect(posX + 21, posY + 2, 4, 10, GxEPD_BLACK);
             // Serial.println("-->[EINK] Drawn rect at " + String(posX + 26) + ", " + String(posY + 2) + " with size 4x10");
         }
     }
+    if (oldBatteryValue == newBatteryValue)
+        return false;
+    else
+        oldBatteryValue = newBatteryValue;
+    return true;
 }
 
-void showCO2(uint16_t co2, int32_t posX, int32_t posY, bool forceRedraw) {
+bool showCO2(uint16_t co2, int32_t posX, int32_t posY, bool forceRedraw) {
     RTC_DATA_ATTR static uint16_t oldCO2Value = 0;
-    if (!forceRedraw && (co2 == oldCO2Value)) return;
-    if ((co2 == 0) || (co2 > 9999)) return;
+    if (!forceRedraw && (co2 <= (oldCO2Value + co2_HYSTERESIS)) && (co2 >= (oldCO2Value - co2_HYSTERESIS))) return false;  // don't update screen CO2 value
+    if ((co2 == 0) || (co2 > 9999)) return false;
 
     display.setRotation(1);
     display.setPartialWindow(0, 0, display.width(), display.height());
+    display.fillRect(posX, posY, display.width(), elementPosition.co2H, GxEPD_WHITE);  // clear previous values
 
-    display.fillRoundRect(0, elementPosition.co2Y, display.width(), elementPosition.bifFontDigitsHeight + 10, 6, GxEPD_WHITE);  // 10 = 2px for top and bottom rectangle borders + 8px for top and bottom margin
+    //    display.fillRoundRect(0, elementPosition.co2Y, display.width(), elementPosition.bifFontDigitsHeight + 10, 6, GxEPD_WHITE);  // 10 = 2px for top and bottom rectangle borders + 8px for top and bottom margin
```

Comment:
So, you don't like the rectangle. 🤔
I think it helps structure the display. Let's discuss it with the Alpha Testers goup!

Reply:
I don't understand... I only replace your fillRoundRect(WHITE) with fillRect(WHITE).

Rectangle is displayed with drawRoundRect(BLACK).